### PR TITLE
[Ldap] Add exception for mapping ldap errors

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #28677   
| License       | MIT
| Doc PR        | 

Maybe we could add more exception code since the list has a lot of errors, maybe we could add a class that maps the error to the right exeptions. see https://www.php.net/manual/en/function.ldap-errno.php

### CODE REVIEW

File: src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php [hunk 1]
```diff
@@ -51,7 +54,19 @@ public function bind($dn = null, $password = null)
         }
 
         if (false === @ldap_bind($this->connection, $dn, $password)) {
-            throw new ConnectionException(ldap_error($this->connection));
+            $error = ldap_error($this->connection);
+            switch (ldap_errno($this->connection)) {
+                case InvalidCredentialsException::LDAP_INVALID_CREDENTIALS:
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php]: these should be private consts on the Connection class to me

File: src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php [hunk 2]
```diff
@@ -51,7 +54,19 @@ public function bind($dn = null, $password = null)
...
+                    throw new InvalidCredentialsException($error);
+                break;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php]: not needed (and wrong identation)

**Review by nicolas-grekas — COMMENTED** [review]
ping @lyrixx since you're mentioned as author

File: src/Symfony/Component/Ldap/Exception/AlreadyExistsException.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Exception;
+
+/**
+ * ConnectionException is throw if binding to ldap can not be established.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Ldap/Exception/AlreadyExistsException.php]: is thrown (missing "n")
+ wrong message (same below)

**Comment by Simperfit** [comment]
@fabpot this is ready

**Review by csarrazi — APPROVED** [review]

**Comment by csarrazi** [comment]
LGTM 👍

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Simperfit.

**Pull request merged**